### DISCUSSION TITLE
Replacing Application Insights with Seq -- Is this realistic yet? 🤔

### DISCUSSION THREAD

**Discussion by jboarman** [datalust/seq-tickets#1460] [Help]
We have a love/hate relationship with Azure's Application Insights.  It provides awesome out-of-the-box integration with cloud resources, such as Azure SQL Databases, such that interactions become automatically profiled.  However, the UX of their platform is just awful and mind-numbing.

Does Seq offer a way out from our dependency on Application Insights in terms of tightly-integrated and automatic logging and profiling of things like DB and HTTP requests?

Here's a comment from someone in our team when I asked this question internally.  I'm posting here just in case things have changed over the last few years.

> Seq is much more just about logs, whereas application insights gives us some built in profiling of each request, each sql query, and each outgoing http call, along with other request/application metrics.
> 
> I kind of hate application insights but it’s definitely been a god send when debugging issues more times than I can count.

Will it ever be possible to rid ourselves of MS' App Insights?  🤔

**Comment by KodrAus** [comment]
Hey @jboarman! :wave:

As a first-party product, App Insights has a bit of a privileged position in the Microsoft ecosystem with out-of-the-box reach into the Azure platform, .NET SDKs, and Visual Studio. You need to do work to _not_ use it. Seq is focused more on developer-driven application diagnostics, so rather than taking a zero-touch approach, we believe in letting developers make informed choices about what diagnostics they surface from the apps they build and maintain, and then execute on those choices as part of their regular development workflow. We believe that results in the most valuable information being there when you really need it (when you have some production issues to diagnose).

Philosophy aside, with `Microsoft.Extensions.Logging` also having direct support in frameworks like ASP.NET and EF you can get automatic logging and profiling of DB and HTTP requests in Seq by adding it as a sink. There are some docs around on stitching these things together:

- [Adding Seq as a MEL sink in ASP.NET apps](https://docs.datalust.co/docs/microsoft-extensions-logging)
- [Logging HTTP requests in .NET 6](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-logging/?view=aspnetcore-6.0)
- [Logging DB requests made by EF](https://docs.microsoft.com/en-us/ef/core/logging-events-diagnostics/extensions-logging)

You certainly can use Seq alongside Application Insights (and the wider Azure Monitor suite) as a more real-time window into the state of your applications running in Azure if you're also getting value in other ways from Application Insights. If you are looking into migrating from Application Insights to Seq this is probably the path I'd suggest first. You might actually find a happy place with both.

Hope that helps!
